### DISCUSSION TITLE
I would like to get alerts when certain links are used

### DISCUSSION THREAD

**Discussion by BillyCroan** [YOURLS/YOURLS#3709] [Ideas]
I have a short link on my yourls instance that I use to send clients to a "get my ip" site quickly.

You know would be even quicker?  If they didn't have to speak the IP to me because I got a little popup on my laptop with their IP when they clicked it.

yes, I can tail the webserver logs.  But is there a more official hook in yourls I could use to get some sort of push notification out of yourls that a link has been clicked and by who?

my instance is personal use only, so i might just turn this on for all clicks out of curiosity!  I suppose that's the case for a lot of users!

**Comment by dgw** [comment]
1. You probably don't _actually_ want a little popup on your laptop every single time a client hits any specific endpoint. There are _a lot_ of bots on the Internet!
2. If you really DO, building it into a YOURLS plugin is going to be much more work than a standalone PHP script. The very basic IP-fetch is one line: \
   `echo ($_SERVER['REMOTE_ADDR']);` \
   Now, you can make that smarter to deal with X-Forwarded-For, IPv6 vs. v4, and so on, but in the end you just need a string representation and something like Pushover.net to send the notification for you.

(Tailing the webserver logs is a great approach, btw. You can get live updates only when actively seeking them, and no notification spam when you aren't waiting for a client to click.)
